src/H5FDsubfiling/H5FDsubfiling.h

```diff
@@ -417,6 +417,91 @@ H5_DLL herr_t H5Pset_fapl_subfiling(hid_t fapl_id, const H5FD_subfiling_config_t
  */
 H5_DLL herr_t H5Pget_fapl_subfiling(hid_t fapl_id, H5FD_subfiling_config_t *config_out);
 
+/**
+ * \ingroup H5VFD
+ *
+ * \brief Retrieve the subfile names for an HDF5 file using the subfiling VFD
+ *
+ * \file_id{file_id}
+ *
+ * \param[out] filenames Pointer to an array of C strings containing the subfile names.
+ *                       Memory is allocated by the function and must be freed by the caller.
+ * \param[out] len       Pointer to the number of subfiles in the \p filenames array.
+ *
+ * \returns \herr_t
+ *
+ * \details H5FDsubfiling_get_file_mapping() retrieves the names of all physical subfiles
+ *          that collectively make up a logical HDF5 file using the subfiling Virtual File
+ *          Driver (VFD). The subfiling VFD distributes file data across multiple subfiles
+ *          to improve parallel I/O performance, particularly systems where metadata operations
+ *          can become a bottleneck.
+ *
+ *          The function returns an array of subfile names corresponding to the physical files
+ *          stored on the file system. Each MPI rank may be responsible for different subfiles,
+ *          and this function provides visibility into which subfiles are associated with the
+ *          calling rank.
+ *
+ *          **Typical use cases include:**
+ *          - **File management**: Understanding the physical storage layout for backup and archival
+ *          - **Performance analysis**: Identifying subfile distribution patterns
+ *          - **File fusion**: Providing subfile lists to tools like h5fuse for recombining files
+ *          - **Debugging**: Troubleshooting subfiling configuration and I/O patterns
+ *          - **Storage optimization**: Analyzing subfile sizes and distribution
+ *
+ * \note  **Memory management**: The caller must call H5free_memory() on each string in the
+ *        \p filenames array, and then call H5free_memory() on the \p filenames array itself:
+ *        \code{.c}
+ *        for (size_t i = 0; i < len; i++) {
+ *            H5free_memory(filenames[i]);
+ *        }
+ *        H5free_memory(filenames);
+ *        \endcode
+ *
+ * \note  **VFD requirement**: This function only works with files that use the subfiling VFD.
+ *        Calling it on files using other VFDs will result in an error.
+ *
+ * \note  **MPI context**: The function returns subfiles associated with the calling MPI rank.
+ *        Different ranks may receive different subfile lists depending on the subfiling
+ *        configuration and I/O concentrator (IOC) assignment.
+ *
+ * \warning **Platform support**: Subfiling is not supported on Windows. This function will
```

Comment:
Is this true?

Reply:
We may just want to refer to this in a generic way, such as mentioning that the function won't be enabled when support for the Subfiling VFD isn't enabled.